pypeit/spectrographs/magellan_fire.py

```diff
@@ -170,121 +213,160 @@ def check_frame_type(self, ftype, fitstbl, exprng=None):
-        return slitmask
+class MagellanFIRELONGSpectrograph(MagellanFIRESpectrograph):
+    """
+    Child to handle Magellan/FIRE high-throughput data
 
-    @property
-    def norders(self):
-        return 22
+    .. note::
+        For FIRE longslit, science data are usually taken with SUTR readout mode with ~600s exposure
+        (at least for quasar hunting people) and the readout noise is ~6 e-
 
-    @staticmethod
-    def slit2order(islit):
+    """
+    def __init__(self):
+        # Get it started
+        super(MagellanFIRELONGSpectrograph, self).__init__()
+        self.spectrograph = 'magellan_fire_long'
+        self.camera = 'FIRE'
+        self.numhead = 1
+        self.detector = [
+                # Detector 1
+                pypeitpar.DetectorPar(
+                            dataext         = 0,
+                            specaxis        = 0,
+                            specflip        = False,
+                            xgap            = 0.,
+                            ygap            = 0.,
+                            ysize           = 1.,
+                            platescale      = 0.15,
+                            darkcurr        = 0.01,
+                            saturation      = 320000., #32000 for low gain, I set to a higher value to keep data in K-band
+                            nonlinear       = 0.875,
+                            numamplifiers   = 1,
+                            gain            = 3.8,
+                            ronoise         = 6.0, # SUTR readout mode with exposure~600s
+                            datasec         = '[5:2044, 900:1250]',
+                            oscansec        = '[:5, 900:1250]'
+                            )]
 
+    def default_pypeit_par(self):
         """
-        Parameters
-        ----------
-        islit: int, float, or string, slit number
-
-        Returns
-        -------
-        order: int
+        Set default parameters.
         """
+        par = pypeitpar.PypeItPar()
+        par['rdx']['spectrograph'] = 'magellan_fire_long'
 
-        if isinstance(islit, str):
-            islit = int(islit)
-        elif isinstance(islit, np.ndarray):
-            islit = islit.astype(int)
-        elif isinstance(islit, float):
-            islit = int(islit)
-        elif isinstance(islit, int):
-            pass
-        else:
-            msgs.error('Unrecognized type for islit')
-
-        orders = np.arange(32, 11, -1, dtype=int)
-        return orders[islit]
-
-    @staticmethod
-    def order_platescale(binning = None):
+        # No overscan
+        for key in par['calibrations'].keys():
+            if 'frame' in key:
+                par['calibrations'][key]['process']['overscan'] = 'none'
+        # Wavelengths
+        # 1D wavelength solution with OH lines
+        par['calibrations']['wavelengths']['rms_threshold'] = 1.0
+        par['calibrations']['wavelengths']['sigdetect']=3
+        par['calibrations']['wavelengths']['fwhm'] = 20
```

Comment:
Is the FWHM really 20 pixels? You are that spectrally oversampled??

Reply:
It is oversampled. I tweaked this parameter a bit, I do need ~20 to get the wavelength solution works.